packages/package.go

```diff
@@ -184,6 +187,15 @@ type Download struct {
 	Type string `config:"type" json:"type" validate:"required"`
 }
 
+type DeploymentModes struct {
+	Default   *DeploymentMode `config:"default,omitempty" json:"default,omitempty" yaml:"default,omitempty"`
+	Agentless *DeploymentMode `config:"agentless,omitempty" json:"agentless,omitempty" yaml:"agentless,omitempty"`
```

Comment:
Not added the `organization`, `division` and `team` fields that can also be set under `agentless` key:

https://github.com/elastic/package-spec/blob/cc5282374a5d062497bf0517d77a449d7176a35e/spec/integration/manifest.spec.yml#L177-L197

Should they be added? Are they necessary for Fleet?

Reply:
According to the description these fields are only intended to be used internally, when the package is installed: https://github.com/elastic/package-spec/issues/795
So I don't think we need to expose them here. cc @seanrathier just in case.

Reply:
They need to be added because Kibana will send those fields to the Agentless API and then the Agentless API will tag the agentless agent deployment with that information.

Does that answer the question?

Reply:
Looking at this comment, I thought those values would be read directly from the manifest in Kibana (Fleet):
https://github.com/elastic/package-spec/issues/795#issuecomment-2337993455

If the package is installed before sending those values to the Agentless API, could those be read from the package manifest? If that is possible, probably it's not needed to add those values to the API.

WDYT ? @seanrathier @jsoriano

Reply:
If I am misunderstanding this request please let me know.

Are we looking to expose `deployment_modes` when searching for packages?  If so,  we don't need to expose the org, division and team in the search.

Reply:
> Are we looking to expose `deployment_modes` when searching for packages? If so, we don't need to expose the org, division and team in the search.

Exactly, I think I didn't give the full context here, sorry. This question was more related to check whether or not those fields should be added in to the search and package API responses from the package registry.

So, I think it's good to proceed as it is the PR now (just exposing if the mode is enabled or not).

Thanks @seanrathier !